Assets/AudioFadeIn.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(AudioSource))]
+public class AudioFadeIn : MonoBehaviour
+{
+        [SerializeField]
+    private int m_FadeInTime = 10;
+    private AudioSource m_AudioSource;
+
+    void Awake()
+    {
+        m_AudioSource = GetComponent<AudioSource>();
+    }
+
+    void Update()
+    {
+        if (m_AudioSource.volume < 1)
+        {
+            m_AudioSource.volume = m_AudioSource.volume + (Time.deltaTime / (m_FadeInTime + 1));
+        }
+        else
+        {
+            Destroy(this);
```

Comment:
Makes sense! This was an idea from the resource where I learned to do audio fading, so I just kept it in because it felt tidy at the time and I hadn't seen that trick before. But, in hindsight, it's really destroying any option to use this game object for more stuff. There's no actual benefit to the destruction. I removed it from the script.